File: examples/ciflyRExamples/tests/testthat/testMinDsep.R
library("here")
source(here("R", "minDsep.R"))

# figure from van der Zander (2020) paper
test_that("Figure 4 left", {
	dirEdges <- rbind(c(1, 3), c(2, 1), c(2, 3), c(3, 4), c(4, 5))
	g <- list("-->" = dirEdges)
	x <- 1
	y <- 5

	minDsep <- findMinSeparator(g, c(x), c(y), c(), seq(5))
	expect_equal(minDsep, c(3))
})

test_that("Figure 4 right", {
	dirEdges <- rbind(c(2, 1), c(2, 3), c(3, 5), c(4, 3), c(4, 5))
	bidirEdges <- rbind(c(1, 3))
	g <- list("-->" = dirEdges, "<->" = bidirEdges)
	x <- 1
	y <- 5

	minDsep <- findMinSeparator(g, c(x), c(y), c(), seq(5))
	expect_equal(sort(minDsep), c(3, 4))
})

File: examples/ciflypy-examples/tests/test_min_dsep.py
from ciflypy_examples.min_dsep import find_min_separator


# figure from van der Zander (2020) paper
def test_figure_4_left():
    # labels = ["X", "V1", "V2", "V3", "Y"]
    admg = {"-->": [(0, 2), (1, 0), (1, 2), (2, 3), (3, 4)]}
    x = 0
    y = 4

    min_dsep = find_min_separator(admg, [x], [y], [], list(range(5)))
    assert min_dsep is not None
    assert set(min_dsep) == {2}


def test_figure_4_right():
    # labels = ["X", V1", "V2", "V3", "Y"]
    admg = {"-->": [(1, 0), (1, 2), (2, 4), (3, 2), (3, 4)], "<->": [(0, 2)]}
    x = 0
    y = 4

    min_dsep = find_min_separator(admg, [x], [y], [], list(range(5)))
    assert min_dsep is not None
    assert set(min_dsep) == {2, 3}
